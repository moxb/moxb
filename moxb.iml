<?xml version="1.0" encoding="UTF-8"?>
<module type="WEB_MODULE" version="4">
  <component name="NewModuleRootManager" inherit-compiler-output="true">
    <exclude-output />
    <content url="file://$MODULE_DIR$">
      <sourceFolder url="file://$MODULE_DIR$/packages/antd/src" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/packages/meteor/src" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/packages/moxb/src" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/packages/react-html/src" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/examples/src" isTestSource="false" />
      <excludeFolder url="file://$MODULE_DIR$/packages/moxb/dist" />
      <excludeFolder url="file://$MODULE_DIR$/packages/meteor/dist" />
      <excludeFolder url="file://$MODULE_DIR$/packages/react-html/dist" />
      <excludeFolder url="file://$MODULE_DIR$/packages/antd/dist" />
      <excludeFolder url="file://$MODULE_DIR$/admin/node-installation" />
      <excludeFolder url="file://$MODULE_DIR$/admin/bin-tools" />
      <excludeFolder url="file://$MODULE_DIR$/examples/dist" />
      <excludeFolder url="file://$MODULE_DIR$/examples/.cache" />
      <excludeFolder url="file://$MODULE_DIR$/coverage" />
    </content>
    <orderEntry type="inheritedJdk" />
    <orderEntry type="sourceFolder" forTests="false" />
  </component>
</module>