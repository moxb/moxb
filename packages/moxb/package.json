{
  "name": "@moxb/moxb",
  "version": "0.2.0-beta.59",
  "description": "MobX data binding",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "main": "dist",
  "typings": "dist/",
  "files": [
    "dist/**/*"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/moxb/moxb.git"
  },
  "author": "moxb (moxb.org)",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/moxb/moxb/issues"
  },
  "homepage": "https://github.com/moxb/moxb#readme",
  "keywords": [
    "mobx",
    "UI"
  ],
  "jest": {
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "jsx",
      "json",
      "node"
    ],
    "transform": {
      "^.+\\.(ts|tsx)$": "ts-jest"
    },
    "testEnvironment": "node",
    "testMatch": [
      "**/src/**/__tests__/*.(ts|tsx)"
    ],
    "modulePaths": [
      "<rootDir>/peer-dependencies/node_modules"
    ]
  },
  "peerDependencies": {
    "mobx": "^5.15.6",
    "tslib": "^2.0.1"
  },
  "devDependencies": {
    "@types/atob": "2.1.2",
    "@types/jest": "26.0.13",
    "@types/lodash.flatten": "4.4.6",
    "@types/urijs": "1.19.11",
    "mobx": "5.13.1",
    "tslib": "1.10.0"
  },
  "dependencies": {
    "btoa-lite": "1.0.0",
    "history": "5.0.0",
    "lodash.flatten": "4.4.0",
    "moment": "2.29.1",
    "urijs": "1.19.2"
  }
}
