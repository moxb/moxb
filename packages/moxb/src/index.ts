export * from './impl';
export * from './types';
export * from './routing';
