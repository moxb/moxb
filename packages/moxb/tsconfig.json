{
    "compilerOptions": {
        "module": "commonjs",
        "target": "ES5",
        "jsx": "react",
        "lib": ["dom","es6","es2015","es2017", "dom.iterable","scripthost"],
        "inlineSourceMap": true,
        "importHelpers":true,
        "experimentalDecorators": true,
        "noImplicitAny": true,
        "noImplicitThis": true,
        "strict": true,
        "outDir": "./dist",
        "declaration": true,
        "skipLibCheck": true,
        "strictNullChecks": true,
        "removeComments": false,
        "noUnusedLocals": true,
        "noUnusedParameters": true,
        "allowUnreachableCode": false,
        "baseUrl": "./src",
        "paths": {
            "*": [
                "*",
                "../../../node_modules/*"
            ]
        }
    },
    "exclude": [
        "./node_modules",
        "./dist"
    ]
}
