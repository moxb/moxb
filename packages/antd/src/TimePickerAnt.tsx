import { FormItemProps } from 'antd/lib/form';
import { observer } from 'mobx-react';
import { CSSProperties } from 'react';
import * as React from 'react';
import { TimePicker } from 'antd';
import { TimePickerProps } from 'antd/lib/time-picker';
import { parseProps } from './BindAnt';
import { Time } from '@moxb/moxb';
import { FormItemAnt, parsePropsForChild } from './FormItemAnt';

export interface BindTimePickerAntProps extends TimePickerProps {
    operation: Time;
    formStyle?: CSSProperties;
}

@observer
export class TimePickerAnt extends React.Component<BindTimePickerAntProps> {
    render() {
        const { operation, invisible, ...props } = parseProps(this.props, this.props.operation);
        if (invisible) {
            return null;
        }
        return (
            <TimePicker
                placeholder={operation.placeholder}
                onChange={(_time: any, timeString: string) => operation.setValue(timeString)}
                {...props as any}
            />
        );
    }
}

@observer
export class TimePickerFormAnt extends React.Component<BindTimePickerAntProps & FormItemProps> {
    render() {
        const { operation, invisible, ...props } = parsePropsForChild(this.props, this.props.operation);
        if (invisible) {
            return null;
        }
        return (
            <FormItemAnt operation={operation} {...this.props as any}>
                <TimePickerAnt operation={operation} {...props} />
            </FormItemAnt>
        );
    }
}
