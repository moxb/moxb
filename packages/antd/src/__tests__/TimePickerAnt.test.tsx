import { TimeImpl } from '@moxb/moxb';
import { shallow } from 'enzyme';
import * as React from 'react';
import { TimePickerAnt, TimePickerFormAnt } from '../TimePickerAnt';
import { shallowMoxbToJson } from './enzymeHelper';

describe('TimePickerAnt', function() {
    it('should render a time picker by default', function() {
        const operation = new TimeImpl({
            id: 'TheId',
            label: 'oh',
            disabled: () => false,
        });
        expect(shallowMoxbToJson(shallow(<TimePickerAnt operation={operation} />))).toMatchSnapshot();
    });
    it('should return null if invisible', function() {
        const operation = new TimeImpl({
            id: 'TheId',
            label: 'oh',
            invisible: () => true,
        });
        expect(shallow(<TimePickerAnt operation={operation} />).type()).toBeNull();
    });
});

describe('TimePickerFormAnt', function() {
    it('should render form item with a time picker by default', function() {
        const operation = new TimeImpl({
            id: 'TheId',
            label: 'oh',
            disabled: () => false,
        });
        expect(shallowMoxbToJson(shallow(<TimePickerFormAnt operation={operation} />))).toMatchSnapshot();
    });
    it('should return null if invisible', function() {
        const operation = new TimeImpl({
            id: 'TheId',
            label: 'oh',
            invisible: () => true,
        });
        expect(shallow(<TimePickerFormAnt operation={operation} />).type()).toBeNull();
    });
});
