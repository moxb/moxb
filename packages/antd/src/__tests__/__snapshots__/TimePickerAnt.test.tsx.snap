// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimePickerAnt should render a time picker by default 1`] = `
<TimePicker
  align={
    Object {
      "offset": Array [
        0,
        -2,
      ],
    }
  }
  data-testid="TheId"
  disabled={false}
  focusOnOpen={true}
  hideDisabledOptions={false}
  id="the_id"
  label="oh"
  onChange={[Function]}
  placement="bottomLeft"
  readOnly={false}
  transitionName="slide-up"
/>
`;

exports[`TimePickerFormAnt should render form item with a time picker by default 1`] = `
<FormItemAnt
  operation="bound TimeImpl"
>
  <TimePickerAnt
    disabled={false}
    id="the_id"
    operation="bound TimeImpl"
    readOnly={false}
  />
</FormItemAnt>
`;
